plugins/catalog/src/plugin.ts

```diff
@@ -35,6 +38,11 @@ export const catalogPlugin = createPlugin({
       deps: { discoveryApi: discoveryApiRef },
       factory: ({ discoveryApi }) => new CatalogClient({ discoveryApi }),
     }),
+    createApiFactory({
+      api: catalogIconResolverRef,
+      deps: { discoveryApi: discoveryApiRef },
```

Comment:
Don't think we need the `discoveryApi` here

Reply:
I just came to the realization that using an API to return react components breaks lazy loading for these components? I don't think that this is an issue here, but it's interesting in the api docs where we use the same pattern. Would be nice to load big libraries like swagger-ui, asyncapi or the graphql viewer lazily.

Reply:
Interesting. I'll need to take a closer look.

Reply:
Am intending to make APIs lazy loaded at some point :>

Would be significantly simpler to implement with suspense though, but who knows when that'll be around 😅 

In general I think it's desirable to avoid having APIs tied to react components and use other extension patterns instead, but in some cases it's prolly the cleanest solution